src/enkf/gridio_fv3reg.f90

```diff
@@ -659,496 +661,1825 @@ subroutine writegriddata(nanal1,nanal2,vars3d,vars2d,n3d,n2d,levels,ndim,vargrid
        call stop2(23)
     endif
     ne = 0
-  ensmemloop: do nanal=nanal1,nanal2
+    ensmemloop: do nanal=nanal1,nanal2
     ne = ne + 1
 
-  backgroundloop: do nb=1,nbackgrounds
-    allocate(workinc3d(nx_res,ny_res,nlevs),workinc3d2(nx_res,ny_res,nlevsp1))
-    allocate(workvar3d(nx_res,ny_res,nlevs))
-    allocate(qworkvar3d(nx_res,ny_res,nlevs))
-    allocate(tvworkvar3d(nx_res,ny_res,nlevs))
+     backgroundloop: do nb=1,nbackgrounds
+      allocate(workinc3d(nx_res,ny_res,nlevs),workinc3d2(nx_res,ny_res,nlevsp1))
+      allocate(workvar3d(nx_res,ny_res,nlevs))
```

Comment:
Please ensure at least two-space indentation for ensmemloop and backgroundloop through out.